Giada: initialize at 0.15.4

Zer0- opened the pull request:
###### Motivation for this change
Giada is not in nixpkgs, it is one of the more capable sound looping programs on linux right now.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Zer0- committed:
New application giada initialized at 0.15.4

Zer0- committed:
use the install command for installPhase

jtojnar commented on pkgs/applications/audio/giada/default.nix:
```diff
@@ -0,0 +1,72 @@
+      '';
+      homepage = https://www.giadamusic.com;
+      platforms = platforms.linux;
+      license = lib.licenses.gpl3;
```
```suggestion
      license = lib.licenses.gpl3Plus;
```

https://github.com/monocasual/giada/blob/251c67c3431b85ef3ac634e7348136cf8843b06d/src/main.cpp#L11-L14

jtojnar commented on pkgs/applications/audio/giada/default.nix:
```diff
@@ -0,0 +1,72 @@
+  src = fetchFromGitHub {
+    owner = "monocasual";
+    repo = "giada";
+    rev = "0e9a7aa5184dc73c6d39e2e5ee6c088c46b28aa6";
```
```suggestion
    rev = "v${version}";
```

jtojnar commented on pkgs/applications/audio/giada/default.nix:
```diff
@@ -0,0 +1,72 @@
+    libpulseaudio
+  ];
+
+  buildPhase = ''
```
You should not be overriding phases unless you know what you are doing. It breaks many features of generic builder like pre/post phase hooks and `configureFlags`.

Instead you should use

```nix
preConfigure = ''
  ${stdenv.shell} ./autogen.sh
'';

configureFlags = [
  "--target=linux"
];
```

The generic builder will take care of running `configure`, `make` and `make install`.

You can also try adding `doCheck = true` to run `make check`.

Also do not forget to add `--enable-system-catch` to `configureFlags`, see https://www.giadamusic.com/documentation-compiling-from-source

jtojnar commented on pkgs/applications/audio/giada/default.nix:
```diff
@@ -0,0 +1,72 @@
+  };
+
+  buildInputs = [
+    autoconf
```
Programs run during build should go to `nativeBuildInputs`.

jtojnar commented on pkgs/applications/audio/giada/default.nix:
```diff
@@ -0,0 +1,72 @@
+}:
+
+stdenv.mkDerivation rec {
+  name = "${pname}-${version}";
```
This is no longer necessary, `mkDerivation` does it by default.

Zer0- committed:
Add recommended fixes

- change license to gplv3+
- don't override buildPhase
- move build tools into nativeBuildInputs
- other minor fixes

Zer0- commented:
@jtojnar OK, Thank you for the feedback. I've made changes towards good practices.

Zer0- commented:
"This PR breaks listing of package outputs after merging. " what does this mean?

aanderse commented:
@GrahamcOfBorg eval

aanderse commented on pkgs/applications/audio/giada/default.nix:
```diff
@@ -0,0 +1,77 @@
+      '';
+      homepage = https://www.giadamusic.com;
+      platforms = platforms.linux;
+      license = lib.licenses.gpl3Plus;
```
remove `lib.`.

aanderse reviewed (changes requested):
@Zer0- looks like an evaluation error. Please correct.

prusnak commented:
Giada has been packaged since cb0b0e93a380d85d901b056b5df3570841899678
